### PR TITLE
Install Posthog snippet

### PR DESCRIPTION
**Pull request by imtuyethan**
- Install Posthog snippet
- Will create events/actions once Faisal completes Jan's new web design

### CODE REVIEW

**Comment by imtuyethan** [comment]
@urmauur 
Hey can you help me check this error, this snippet works on my local but not sure why it is built unsuccessfully 

Screenshot 2023-11-09 at 2 29 54 PM

File: docs/src/pages/index.js [hunk 1]
```diff
@@ -17,6 +17,9 @@ import {
 } from "@heroicons/react/24/outline";
 
 import ThemedImage from "@theme/ThemedImage";
+import posthog from 'posthog-js'
+
+posthog.init('phc_cJ95zWbMwdef6nVasPCoSNOvV8lUcL5IykIYOoyGXVm', { api_host: 'https://app.posthog.com' })
```
Comment (louis-jan) [inline docs/src/pages/index.js:22]: cc @hiento09 on Github Secret

**Comment by hiento09** [comment]
image

**Review by louis-jan — APPROVED** [review]
LGTM

**Comment by dan-jan** [comment]
> @urmauur Hey can you help me check this error, this snippet works on my local but not sure why it is built unsuccessfully
> 
> Screenshot 2023-11-09 at 2 29 54 PM

I've also experienced this issue, but it doesn't seem to affect building on Prod

**Comment by dan-jan** [comment]
@imtuyethan @hiento09 I encountered this issue when running `yarn start`. Is the common practice to disable GTM in dev environment? Or do we have a dev branch GTM_ID?

image

**Comment by dan-jan** [comment]
> @dan-jan we move to env, will DM you regarding `.env`

I think my concern is with contributors who might want to git clone Docs and add to it. We should not require them to have a `GTM_ID` filled from `.env`, otherwise it adds difficulty and will deter contributions. 

Is it still best practice in 2023 to check for `window.hostname` before firing GTM/Google Analytics? 
https://stackoverflow.com/questions/40297763/how-to-disable-google-analytics-on-localhost

**Comment by dan-jan** [comment]
@hiento09 @louis-jan @imtuyethan @urmauur Btw, for Docs, I actually would recommend having hard-coded Posthog and GTM IDs. 

- There should only be one Jan.ai docs site 
- `.env` injections are better for self-hosted apps, which are defined for GTM, Posthog to be different
- We can track when lazy forkers just fork and deploy our site (it will happen)

Hard-coding Analytics IDs ironically helps us find out when lazy forkers just deploy a docs site. I don't intend to prevent that, but it would be nice to be able to track that from our analytics. We can then see traffic stats by domain (including localhost, as a measure of dev documentation activity).

**Comment by urmauur** [comment]
Hi @dan-jan not sure if we hardcode the GTM-ID, when someone fork the Jan docs and deploy to their web example, what happen if they still using our GTM-ID, but if we worried someone to contribute to Jan docs, i think we can add default something like this 

`googleTagManager: { containerId: process.env.GTM_ID || "XXX", },`
and 
`{
        apiKey: process.env.POSTHOG_PROJECT_API_KEY || "XXX",
        appUrl: process.env.POSTHOG_APP_URL || "XXX", // optional
        enableInDevelopment: false, // optional
}
`

so even someone missing `.env` still can run `yarn start`

cc @louis-jan

**Comment by dan-jan** [comment]
Sounds good! My first objective is to make sure someone can contribute, without a `.env`. The "lazy forker" is less important.

**Pull request merged**